File: LeetCode/401-500/452. Minimum Number of Arrows to Burst Balloons/src/Solution.java
import java.util.Arrays;
import java.util.Comparator;

class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, Comparator.comparingInt(x -> x[1]));
        int count = 1;
        int end = points[0][1];
        for (int i = 1; i < points.length; i++) {
            if (points[i][0] > end) {
                count++;
                end = points[i][1];
            }
        }
        return count;
    }
}

File: LeetCode/401-500/452. Minimum Number of Arrows to Burst Balloons/Solution.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end(), [](vector<int>& a, vector<int>& b) {
            return a[1] < b[1];
        });
        int count = 1;
        int prevEnd = points[0][1];
        for (int i = 1; i < points.size(); ++i) {
            if (points[i][0] > prevEnd) {
                count++;
                prevEnd = points[i][1];
            }
        }
        return count;
    }
};
